File: bestia-messages/src/main/java/net/bestia/messages/dialog/DialogMessage.java
package net.bestia.messages.dialog;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import net.bestia.messages.Message;

public class DialogMessage extends Message {

	private static final long serialVersionUID = 1L;
	public static final String MESSAGE_ID = "ui.dialog";
	
	@JsonProperty("n")
	private final List<DialogNode> nodes;
	
	public DialogMessage(List<DialogNode> nodes) {
		this.nodes = Collections.unmodifiableList(new ArrayList<>(nodes));
	}

	@Override
	public String getMessageId() {
		return MESSAGE_ID;
	}

	@Override
	public String getMessagePath() {
		return getClientMessagePath();
	}

}

File: bestia-messages/src/main/kotlin/net/bestia/messages/ui/DialogMessage.kt
package net.bestia.messages.ui

import net.bestia.messages.AccountMessage

/**
 * This message will trigger the client to open a dialog NPC box.
 *
 * @author Thomas Felix
 */
data class DialogMessage(
    override val accountId: Long,
    private val nodes: List<DialogNode>
) : AccountMessage
